lib/hawkular/operations/operations_api.rb

```diff
@@ -215,7 +217,10 @@ def check_pre_conditions(hash = {}, params = [], &callback)
       fail 'callback must have the perform method defined. include Hawkular::Operations' unless
           callback.nil? || callback.respond_to?('perform')
       params.each do |property|
-        fail "Hash property #{property} must be specified" if hash[property].nil?
+        next unless hash[property].nil?
+        err_message = "Hash property #{property} must be specified"
+        fail(ArgumentError, err_message) if callback.nil?
```

Comment:
See below - if all callbacks are missing or only the error callback is missing, we need to fail, saying so.
Once the error callback is present, all failures need to be returned via the error callback.